rootsanalyzer/rootset_id.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootsanalyzer
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/monologue/storage"
+)
+
+// SHAIDMaker is SHA256-based implementation IDMaker for root certificates ID generation.
+type SHAIDMaker struct {
+}
+
+func (m *SHAIDMaker) GenerateCertID(root *x509.Certificate) ([32]byte, error) {
+	if root == nil {
+		return [32]byte{}, fmt.Errorf("Unable to generate root-ID for nil")
+	}
+	return sha256.Sum256(root.Raw), nil
+}
+
+func (m *SHAIDMaker) GenerateSetID(roots []*x509.Certificate) (*storage.RootSetID, error) {
```

Comment:
Strings are usually passed/returned by value (they aren't copied, since they're immutable).

```suggestion
func (m *SHAIDMaker) GenerateSetID(roots []*x509.Certificate) (storage.RootSetID, error) {
```

Reply:
Also, this doesn't implement `storage.IDMaker` unless it returns `RootSetID` by value.

Reply:
Outdated since interface is removed.

Reply:
Why are you returning it as a pointer?

Reply:
Changed